test/managedwriter/writer_client_test.ts

```diff
@@ -0,0 +1,371 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ** This file is automatically generated by gapic-generator-typescript. **
+// ** https://github.com/googleapis/gapic-generator-typescript **
+// ** All changes to this file may be overwritten. **
+
+import '../../../package.json';
+import * as gax from 'google-gax';
+import * as protos from '../../protos/protos';
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import {describe, it} from 'mocha';
+import * as protobufjs from 'protobufjs';
+import * as bigquerywriterModule from '../../src';
+import {CustomerRecordMessage} from '../managedwriter/test_protos';
+import * as customer_record from './test_protos/customer_record';
+import customerRecordProtoJson = require('../managedwriter/test_protos/customer_record.json');
+
+import {ClientOptions} from 'google-gax';
+
+type WriteStream = protos.google.cloud.bigquery.storage.v1.IWriteStream;
+type ProtoData =
+  protos.google.cloud.bigquery.storage.v1.AppendRowsRequest.IProtoData;
+type ProtoSchema = protos.google.cloud.bigquery.storage.v1.IProtoSchema;
+type ProtoDescriptor = protos.google.protobuf.IDescriptorProto;
+type IInt64Value = protos.google.protobuf.IInt64Value;
+type AppendRowsResponse =
+  protos.google.cloud.bigquery.storage.v1.IAppendRowsResponse;
+const type = protos.google.protobuf.FieldDescriptorProto.Type;
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+
+describe('managedwriter.WriterClient', () => {
```

Comment:
I think for those tests, would be nice to test the backend behavior in different scenarios, not sure if we will be feasible to mock everything in. So as part of the initialization process, we need to set up a test dataset and table. For testing purposes I ended up creating them manually here, but for CI we might need to automate that, similarly on how we do for Samples testing.